src/Microsoft.Data.SqlClient/src/Microsoft/Data/SqlClient/SqlBulkCopy.cs

```diff
@@ -3002,14 +3049,27 @@ private void WriteToServerInternalRestAsync(CancellationToken cts, TaskCompletio
                         TaskCompletionSource<object> cancellableReconnectTS = new TaskCompletionSource<object>();
                         if (cts.CanBeCanceled)
                         {
-                            regReconnectCancel.Value = cts.Register(() => cancellableReconnectTS.TrySetCanceled());
+                            regReconnectCancel.Value = cts.Register(
```

Comment:
These methods, taken from netcore implementation, are being used since they use static lambdas and are slightly more performant.